common/app/routes/challenges/components/Classic-Modal.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { PropTypes } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import PureComponent from 'react-pure-render/component';
+
+export default class ClassicModal extends PureComponent {
+  static propTypes = {
+    buttonText: PropTypes.string,
+    children: PropTypes.node,
+    customClass: PropTypes.string,
+    keyBindings: PropTypes.func,
+    modalShow: PropTypes.bool.isRequired,
+    modalTitle: PropTypes.string,
+    onModalClick: PropTypes.func,
+    onModalExit: PropTypes.func
+  }
+
+  onClickHandler() {
+    if (this.props.onModalClick) {
+      return this.props.onModalClick();
+    }
+    throw new Error('Expected an "onClick" handler for "onModalClick"');
+  }
+  noOp() {
+    return false;
+  }
+  render() {
+    let {
+      buttonText,
+      customClass,
+      keyBindings,
+      modalShow,
+      modalTitle,
+      onModalExit
+    } = this.props;
+    let _buttonText = (buttonText) ? buttonText : 'Submit',
+        _customClass = (customClass) ? customClass : '',
+        _modalTitle = (modalTitle) ? modalTitle : '',
+        _onModalExit = (onModalExit) ? onModalExit : this.noOp.bind(this),
```

Comment:
A noop can be a plain function, it does not need to be a method on a class.
styl-bind-in-render: You shouldn't use the bind method in a render as this creates a new function and causes this components children to re-render without needing to.